### PR TITLE
[SD-3166] - Use the 'en' dictionary if the language of story is 'en-US' or 'en-AU'

### PR DESCRIPTION
**Pull request by akintolga**

### CODE REVIEW

**Comment by akintolga** [comment]
This PR replaces my earlier (closed few weeks ago) PR #1257. 

With this change system will use the specific dictionary (i.e. en-AU) if it exists otherwise it will use base dictionary (en) if it exists.  If the story is set to have en as the language then it will use 'en', not  'en-AU' even if it exists. So we expect the base dictionary (en) is set up to reflect the main language the system will use.

@petrjasek I have done some refactoring in the suggest endpoint as well.

File: server/apps/dictionaries/service.py [hunk 1]
```diff
@@ -111,8 +137,8 @@ def get_model_for_lang(self, lang):
         :param lang: language code
         """
         model = {}
-        lookup = {'$and': [{'language_id': lang}, {'is_active': {'$in': ['true', None]}}]}
-        dicts = self.get(req=None, lookup=lookup)
+        dicts = self.get_dictionaries(lang)
```
Comment (mugurrus) [inline server/apps/dictionaries/service.py]: I suggest caching dictionaries and models. This takes a lot of processing and the spelling suggestion requests should happen quite often.

**Comment by mugurrus** [comment]
:+1: (except the minor comment)

File: client/app/scripts/superdesk-dictionaries/dictionaries.js [hunk 2]
```diff
@@ -96,14 +96,19 @@
          *
          * @param {string} lang
          */
```
Comment (plamut) [inline client/app/scripts/superdesk-dictionaries/dictionaries.js]: Docstring needs an update here (new parameter).

File: server/apps/dictionaries/service.py [hunk 3]
```diff
@@ -103,6 +103,32 @@ def on_fetched(self, result):
...
+        """
+        languages = [{'language_id': lang}]
+        base_language = self.get_base_language(lang)
+        if base_language:
+            languages.append({'language_id': base_language})
+
+        lookup = {'$and': [{'$or': languages},
+                           {'$or': [{'is_active': {'$exists': 0}}, {'is_active': 'true'}]}]}
+        dicts = list(self.get(req=None, lookup=lookup))
+        langs = [d['language_id'] for d in dicts]
```
Comment (plamut) [inline server/apps/dictionaries/service.py]: `langs` could be a `set()` here, but it probably doesn't matter because the expected list length is small.

**Comment by plamut** [comment]
Looks pretty good all in all!

:+1:

**Pull request merged**